codex/stores/repostore.nim

```diff
@@ -82,6 +77,62 @@ func available*(self: RepoStore): uint =
 func available*(self: RepoStore, bytes: uint): bool =
   return bytes < self.available()
 
+proc encode(cidAndProof: (Cid, MerkleProof)): seq[byte] =
+  let 
+    (cid, proof) = cidAndProof
+    cidBytes = cid.data.buffer
+    proofBytes = proof.encode
+
+  var buf = newSeq[byte](1 + cidBytes.len + proofBytes.len)
+
+  buf[0] = cid.data.buffer.len.byte # cid shouldnt be more than 255 bytes?
+  buf[1..cidBytes.len] = cidBytes
+  buf[cidBytes.len + 1..^1] = proofBytes
+
+  buf
+
+proc decode(_: type (Cid, MerkleProof), data: seq[byte]): ?!(Cid, MerkleProof) =
+  let cidLen = data[0].int
+
+  let 
+    cid = ? Cid.init(data[1..cidLen]).mapFailure
+    proof = ? MerkleProof.decode(data[cidLen + 1..^1])
+  
+  success((cid, proof))
+
+method putBlockCidAndProof*(
+  self: RepoStore,
+  treeCid: Cid,
+  index: Natural,
+  blockCid: Cid,
+  proof: MerkleProof
+): Future[?!void] {.async.} =
+  ## Put a block to the blockstore
+  ##
+
+  without key =? createBlockCidAndProofMetadataKey(treeCid, index), err:
```

Comment:
I don't like such long names either, but I like consistency in a codebase. Procs above this one are of similar length:
```nim
proc createBlockExpirationMetadataKey
proc createBlockExpirationMetadataQueryKey
proc createBlockCidAndProofMetadataKey
```

I could shorten it up to `forCidAndProof` and it would be used like this `Key.forCidAndProof(...)` but it would be different than other procs, so I think I should refactor other procs as well, to keep it consistent. Wdyt?

Reply:
Heh, yeah - sorry for bikeshedding, there is already a lot going on here and we can certainly leave it for another PR. That said, I like the above proposal.